src/scenes/UserGalleryPage/UserGalleryCollections.tsx

```diff
@@ -115,19 +98,68 @@ function UserGalleryCollections({ galleryRef, queryRef, mobileLayout }: Props) {
     return <EmptyGallery message={emptyGalleryMessage} />;
   }
 
+  const rowRenderer = ({ index, key, parent, style }: ListRowProps) => {
+    const collection = collectionsToDisplay[index];
+    return (
+      <CellMeasurer
+        cache={cache.current}
+        columnIndex={0}
+        rowIndex={index}
+        key={key}
+        parent={parent}
+      >
+        {({ registerChild, measure }) => {
+          return (
+            // @ts-ignore
+            <div ref={registerChild} key={key} style={style}>
+              <UserGalleryCollection
+                queryRef={query}
+                collectionRef={collection}
+                mobileLayout={mobileLayout}
+                onLoad={measure}
+              />
+              <Spacer height={48} />
+            </div>
+          );
+        }}
+      </CellMeasurer>
+    );
+  };
+
   return (
     <StyledUserGalleryCollections>
       <Spacer height={isMobile ? 48 : 80} />
-      {collectionsToDisplay.slice(0, numRenderedCollections).map((collection) => (
-        <Fragment key={collection.id}>
-          <UserGalleryCollection
-            queryRef={query}
-            collectionRef={collection}
-            mobileLayout={mobileLayout}
-          />
-          <Spacer height={48} />
-        </Fragment>
-      ))}
+      <WindowScroller>
+        {({ height, registerChild, scrollTop, onChildScroll }) => (
+          <AutoSizer disableHeight>
+            {({ width }) => (
+              <div ref={registerChild}>
+                <List
+                  ref={listRef}
+                  autoHeight
+                  width={width}
+                  height={height}
+                  onScroll={onChildScroll}
+                  rowHeight={cache.current.rowHeight}
+                  rowCount={numCollectionsToDisplay}
+                  scrollTop={scrollTop}
+                  deferredMeasurementCache={cache.current}
+                  rowRenderer={rowRenderer}
+                  overscanIndicesGetter={({
```

Comment:
Not really sure why my `overscanRowCount` didnt work here. Replace it with snippet to render some buffer to avoid flickering.

[Similar issue](https://github.com/bvaughn/react-virtualized/issues/1582)